vision_evaluation/prediction_filters.py

```diff
@@ -18,33 +18,42 @@ def identifier(self):
 
 class TopKPredictionFilter(PredictionFilter):
 
-    def __init__(self, k: int):
+    def __init__(self, k: int, prediction_mode='prob'):
         """
         Args:
             k: k predictions with highest confidence
+            prediction_mode: can be 'indices' or 'prob', indicating whether the predictions are a set of class indices or predicted probabilities.
         """
         assert k >= 0
+        assert prediction_mode == 'prob' or prediction_mode == 'indices', f"Prediction mode {prediction_mode} is not supported!"
 
+        self.prediction_mode = prediction_mode
         self.k = k
 
     def filter(self, predictions, return_mode):
         """ Return k class predictions with highest confidence.
         Args:
-            predictions: predicted probabilities of N samples belonging to C classes. Shape (N, C)
+            predictions:
+                when 'prediction_mode' is 'prob', refers to predicted probabilities of N samples belonging to C classes. Shape (N, C)
+                when 'prediction_mode' is 'indices', refers to indices of M highest confidence of C classes in descending order, for each of the N samples. Shape (N, M)
             return_mode: can be 'indices' or 'vec', indicating whether return value is a set of class indices or 0-1 vector
 
         Returns:
             k labels with highest probabilities, for each sample
         """
 
         k = min(predictions.shape[1], self.k)
-        if k == 0:
-            top_k_pred_indices = np.array([[] for i in range(predictions.shape[1])], dtype=int)
-        elif k == 1:
-            top_k_pred_indices = np.argmax(predictions, axis=1)
-            top_k_pred_indices = top_k_pred_indices.reshape((-1, 1))
+
+        if self.prediction_mode == 'prob':
+            if k == 0:
+                top_k_pred_indices = np.array([[] for i in range(predictions.shape[1])], dtype=int)
+            elif k == 1:
+                top_k_pred_indices = np.argmax(predictions, axis=1)
+                top_k_pred_indices = top_k_pred_indices.reshape((-1, 1))
+            else:
+                top_k_pred_indices = np.argpartition(predictions, -k, axis=1)[:, -k:]
         else:
-            top_k_pred_indices = np.argpartition(predictions, -k, axis=1)[:, -k:]
+            top_k_pred_indices = predictions[:, :k]
```

Comment:
What happens when k > M? 

Can we add a unit test for this filter? Thanks

Reply:
k>M has already be considered in https://github.com/microsoft/vision-evaluation/blob/0f34dcbc7ebd037357698808d056f0dba442f03e/vision_evaluation/prediction_filters.py#L40

The logic is the same with the case when the prediction is prob. The dimension of prob vector C can also be smaller than k.

Reply:
Unit test has already been added in `test_evaluator.py`